index.js

```diff
@@ -297,7 +297,7 @@ class Hyperbee extends ReadyResource {
 
     if (this._onappendBound) {
       this.core.on('append', this._onappendBound)
-      if (this.core.isAutobase) this.core.on('truncate', this._onappendBound)
+      this.core.on('truncate', () => this._onappendBound(true))
```

Comment:
use two different functions, this is relying on append on the core not having any args which will fail at some point